disco_aws_automation/disco_vpc.py

```diff
@@ -156,20 +161,22 @@ def fetch_environment(cls, vpc_id=None, environment_name=None):
         """
         Returns an instance of this class for the specified VPC, or None if it does not exist
         """
-        vpc_conn = VPCConnection()
-        if vpc_id:
-            vpc = vpc_conn.get_all_vpcs(vpc_ids=[vpc_id])
-        elif environment_name:
-            vpc = vpc_conn.get_all_vpcs(filters={"tag:Name": environment_name})
-        else:
-            raise VPCEnvironmentError("Expect vpc_id or environment_name")
-
-        if vpc:
-            vpc = vpc[0]
-            return cls(vpc.tags["Name"], vpc.tags["type"], vpc)
-        else:
+        client = boto3.client('ec2')
+        try:
+            if vpc_id:
+                vpc = client.describe_vpcs(
+                    Filters=[{'Name': 'vpc-id', 'Values': [vpc_id]}])['Vpcs'][0]
+            elif environment_name:
+                vpc = client.describe_vpcs(
+                    Filters=[{'Name': 'tag:Name', 'Values': [environment_name]}])['Vpcs'][0]
+            else:
+                raise VPCEnvironmentError("Expect vpc_id or environment_name")
+        except IndexError:
```

Comment:
This try block is bit sketchy, is it meant to catch the [0]? There are a lot of statements here, how are we sure we are not catching stray exceptions?